src/main/java/org/jabref/gui/copyfiles/CopyFilesTask.java

```diff
@@ -0,0 +1,115 @@
+package org.jabref.gui.copyfiles;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiFunction;
+
+import javafx.concurrent.Task;
+
+import org.jabref.Globals;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.OS;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.util.OptionalUtil;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class CopyFilesTask extends Task<List<CopyFilesResultItemViewModel>> {
+
+    private static final Log LOGGER = LogFactory.getLog(ExportLinkedFilesAction.class);
+    private static final String LOGFILE = "exportLog.log";
```

Comment:
I am unsure if this logfile is needed.

If it is, I would encode a timestamp into the filename to be able to distinguish several exports into the same folder.